content/blog/wallet-descriptors.md

```diff
@@ -0,0 +1,164 @@
+    "pubkey": "0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9",
+    "ischange": false,
+    "timestamp": 1654093495,
+    "hdkeypath": "m/0'/0'/1'",
+    "hdseedid": "50b54a741dbc3b120280391ecba28d657e137bce",
+    "hdmasterfingerprint": "dc711bf8",
+    "labels": [""]
+}
+```
+
+We can get more information about the descriptor in the result above by running the following command:
+
+`bitcoin-cli getdescriptorinfo <descriptor>`
+
+Using the descriptor in the JSON output above, I get the following result.
+
+```json
+{
+  "descriptor": "wpkh([dc711bf8/0'/0'/1']0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9)#yr975l8h",
+  "checksum": "yr975l8h",
+  "isrange": false,
+  "issolvable": true,
+  "hasprivatekeys": false
+}
+```
+
+A script is solvable according to Core, if given the private key and the descriptor, the wallet software can generate a scriptsig/witness to spend its funds. `isrange` denotes if it is a range descriptor meaning it is possible to generate other descriptors from it. Range descriptors usually end their derivation path with `/*`. For example, `wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/0'/0'/0/*)` is a range descriptor.
+
+The `getdescriptorinfo` command can also be used to derive the checksum of a descriptor if there you get one without a checksum. If we run the previous command, this time without adding the checksum at the end, Core will compute the checksum for us in the outputs.
+
+`bitcoin-cli getdescriptorinfo "wpkh([dc711bf8/0'/0'/1']0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9)"`
+
+The output derives the checksum for us. See the JSON result below:
+
+```json
+{
+  "descriptor": "wpkh([dc711bf8/0'/0'/1']0303207b2fd5b151397949aa9d6f9ff8eccd3bb97aa05192ced942b17bbd45e9d9)#yr975l8h",
+  "checksum": "yr975l8h",
+  "isrange": false,
+  "issolvable": true,
+  "hasprivatekeys": false
+}
+```
+
+Aside from getting information about a descriptor, we can derive an address from a
+descriptor. To do that, use the following command:
+
+`bitcoin-cli deriveaddresses <descriptor>`
+
+Using the descriptor above, we get the following address:
+
+```json
+[
+  "bcrt1q2e4g8cetupxes7l95qstpxq8qnpml9ev6gjle2"
+]
+```
+
+Finally, it is possible to import descriptors into another wallet using the `importdescriptors` command. 
+
+`bitcoin-cli -rpcwallet=vladitest importdescriptors '[{ "desc": "wpkh([0ee8ad5c/84'"'"'/1'"'"'/0'"'"'/0/0]0384fa1d3495ae000e6da98daef1b68dfb0730bd1275f2ede4342b013ed8cd15a2)#y8j0e2j6", "timestamp":"now", "label": "" }]'`
+
+Notice the `'"'"'` used to denote the hardened paths in the descriptor, consider this as an escape character. It is also worth mentioning that if you are importing a descriptor with public keys, you 
+will need to create a wallet with private keys disabled by passing `disable_private_keys=true` as an argument.
```

Comment:
So to expand, you could mention that this implies that Bitcoin Core does not handle wallets which include _some_ watch-only addresses and _some_ private key-backed addresses. I think this is done so that the wallet doesn't generate a new address from the watch-only descriptor when you meant to generate an address you had private keys for, and then you find out funds received there are lost/not spendable...